src/rpc/types.rs

```diff
@@ -0,0 +1,85 @@
+use derivative::Derivative;
+use reth::primitives::{Address, Bloom, Bytes, H256, U256};
+use reth::rpc::types::{ExecutionPayload, ExecutionPayloadV1, ExecutionPayloadV2, Withdrawal};
+use serde::{Deserialize, Serialize};
+use serde_this_or_that::as_u64;
+
+/// Structure to deserialize execution payloads sent according to the builder api spec
+/// Numeric fields deserialized as decimals (unlike crate::eth::engine::ExecutionPayload)
+#[derive(Derivative)]
+#[derivative(Debug)]
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[allow(missing_docs)]
```

Comment:
Why do you need new types here? Seems like maintain headache, do we need to make the deser logic better?

Reply:
This is really just because of difference in the formatting of the current api that we are using (snake_case and decimal encoded numbers). 
I agree this is a headache and we should probably just use the existing formatting (camelCase and hex numbers), in which case we can drop these additional types and just use the existing `ExecutionPayload` struct.

@alextes What do you think ?

Reply:
If we do that I could probably also remove [these re-exports](https://github.com/paradigmxyz/reth/pull/4783/files#diff-ab38c46fc23649346bb86710f377a3254c8920e36a3ecea20238546c32023fbfR24) from my pr against the reth repo.

Reply:
I opened an issue for this, and suggest doing this in a separate PR:
https://github.com/ckoopmann/reth-block-validator/issues/2